content/api/commands/origin.md

```diff
@@ -0,0 +1,512 @@
+origin callback, otherwise a cross-domain error will be thrown.
+
+```js
+// Button in primary origin goes to https://www.acme.com
+cy.contains('button', 'Go').click()
+
+cy.origin('www.acme.com', () => {
+  // No cy.visit is needed as the button brought us here
+  cy.get('h1').contains('ACME CORP')
+})
+```
+
+### Navigating to multiple secondary origins in succession
+
+Callbacks may **not** themselves contain `cy.origin()` calls, so when visiting
+multiple origins, do so at the top level of the test.
+
+```js
+cy.origin('foo.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'foo.com')
+})
+
+cy.origin('bar.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'bar.com')
+})
+
+cy.origin('baz.com', () => {
+  cy.visit('/')
+  cy.url().should('contain', 'baz.com')
+})
+```
+
+<Alert type="info">
+
+A future version of Cypress will allow the use of nested `cy.origin()` calls.
+
+</Alert>
+
+### SSO login custom command
+
+A very common requirement is logging in to a site before running a test. If
+login itself is not the specific focus of the test, it's good to encapsulate
+this functionality in a `login`
+[custom command](/api/cypress-api/custom-commands) so you don't have to
+duplicate this login code in every test. Here's an idealized example of how to
+do this with `cy.origin()`.
+
+**<Icon name="exclamation-triangle" color="#f0ad4e"></Icon> Inefficient Usage**
+
+```js
+Cypress.Commands.add('login', (username, password) => {
+  // Remember to pass in dependencies via `args`
+  const args = { username, password }
+  cy.origin('my-auth.com', { args }, ({ username, password }) => {
+    // Go to https://auth-provider.com/login
+    cy.visit('/login')
+    cy.contains('Username').find('input').type(username)
+    cy.contains('Password').find('input').type(password)
+    cy.get('button').contains('Login').click()
+  })
+  // Confirm we're back at the primary origin before continuing
+  cy.url().should('contain', '/home')
+})
+```
+
+Having to go through an entire login flow before every test is not very
+performant. Up until now you could get around this problem by putting login code
+in the first test of your file, then performing subsequent tests reusing the
+same session.
+
+However, once the `experimentalSessionAndOrigin` flag is activated this is no
+longer possible, as all session state is now cleared between tests. So to avoid
+this overhead we have added the [`cy.session()`](/api/commands/session) command,
+which allows you to easily cache session information and reuse it across tests.
+So now let's enhance our custom login command with `cy.session()` for a complete
+syndicated login flow with session caching and validation. No mocking, no
+workarounds, no third-party plugins!
+
+```js
+Cypress.Commands.add('login', (username, password) => {
+  const args = { username, password }
+  cy.session(
+    // Username & password can be used as the cache key too
```

Comment:
We probably shouldn't recommend including password in the session ID since this could leak private/secret information in the Command Log & in test recordings.